orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -789,7 +808,210 @@ static Reader newFileReader(InputFile file, Configuration config) {
     ReaderOptions readerOptions = OrcFile.readerOptions(config).useUTCTimestamp(true);
     if (file instanceof HadoopInputFile) {
       readerOptions.filesystem(((HadoopInputFile) file).getFileSystem());
+    } else {
+      readerOptions.filesystem(new InputFileSystem(file)).maxLength(file.getLength());
```

Comment:
`maxLength` is used to avoid calls to get status? If so, we should add a comment.

Reply:
Added comment to clarify this.